app/cdap/components/WrangleHome/Components/OngoingDataExploration/index.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from 'react';
+import { Box, Grid, Typography } from '@material-ui/core/';
+import ExplorationCardStyles from './styles';
+import { updatedData } from './utils';
+import MyDataPrepApi from 'api/dataprep';
+import { Link } from 'react-router-dom';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+
+const OngoingDataExploration = () => {
+  const classes = ExplorationCardStyles();
+  const [ongoingExpDatas, setOngoingExpDatas] = useState<any>([]);
+  const [finalArray, setFinalArray] = useState([]);
+
+  const getOngoingData = () => {
+    MyDataPrepApi.getWorkspaceList({
+      context: 'default',
+    }).subscribe((res) => {
+      res.values.forEach((item) => {
+        const params = {
+          context: 'default',
+          workspaceId: item.workspaceId,
+        };
+        const requestBody = {
+          directives: item.directives,
+          limit: 1000,
+          insights: {
+            name: item.name,
+            workspaceName: item.workspaceName,
+            path: item?.sampleSpec?.path,
+            visualization: {},
+          },
+        };
+        MyDataPrepApi.execute(params, requestBody).subscribe((response) => {
+          let dataQuality = 0;
+          response.headers.forEach((head) => {
+            const general = response.summary.statistics[head].general;
+            const { empty: empty = 0, 'non-null': nonEmpty = 100 } = general;
+            const nonNull = Math.floor((nonEmpty - empty) * 10) / 10;
+            dataQuality = dataQuality + nonNull;
+          });
+
+          const totalDataQuality = dataQuality / response.headers.length;
+
+          setOngoingExpDatas((current) => [
+            ...current,
+            {
+              connectionName:
+                item?.sampleSpec?.connectionName === undefined
+                  ? 'Upload'
+                  : item?.sampleSpec?.connectionName,
+              workspaceName: item.workspaceName,
+              recipeSteps: item.directives.length,
+              dataQuality: totalDataQuality,
+            },
+          ]);
+        });
+      });
+    });
+  };
+
+  useEffect(() => {
+    getOngoingData();
+  }, []);
+
+  useEffect(() => {
+    const final = updatedData(ongoingExpDatas);
+    setFinalArray(final);
+  }, [ongoingExpDatas]);
+
+  return (
+    <Box data-testid="ongoing-data-explore-parent">
+      {finalArray.map((item) => {
+        return (
+          <Link
+            to={`/ns/${getCurrentNamespace()}/wrangler-grid/:${`${'testDataset'}`}`}
+            style={{ textDecoration: 'none' }}
+          >
+            <Grid container className={classes.gridContainer}>
+              {item.map((eachItem) => {
+                switch (eachItem.type) {
```

Comment:
Move this out to a function or component? To improve readability.

Reply:
Moved this out to the component folder outside WrangleHome Component